src/plugins/credential_management/server/routes/create.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Any modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+import { IRouter } from '../../../../core/server';
+import { schema } from '@osd/config-schema';
+import { createHandler } from '../credential';
+
+export function registerCreateRoute(router: IRouter) {
+    router.post(
+        {
+          path: '/api/credential_management/create',
+          validate: {
+            body: schema.object({
+                credential_name: schema.string(),
+                credential_type: schema.oneOf([
+                  schema.literal('basic_auth'),
+                  schema.literal('aws_iam_credential'),
+                ]),
+                basic_auth_credential_JSON: schema.maybe(schema.object({
+                  user_name: schema.string(),
+                  password: schema.string()
+                })),
+                aws_iam_credential_JSON: schema.maybe(schema.object({
+                  aws_iam_credential: schema.string()
```

Comment:
if we are talking about aws credential 

should it contains

aws_access_key_id = YOUR_AWS_ACCESS_KEY_ID
aws_secret_access_key = YOUR_AWS_SECRET_ACCESS_KEY

and region as optional?